src/test/scala/freeeventsourcing/syntax/ProcessSyntaxTests.scala

```diff
@@ -0,0 +1,241 @@
+  "ProcessSyntax.firstOf " should " accept a single event from an aggregate" in {
+    val r = firstOf(_.
+      from(Account.Id(1)).on[Opened].event)
+    "r : ProcessMonad[Opened :+: CNil]" should compile
+  }
+
+  "ProcessSyntax.firstOf " should " accept a two event from different aggregates" in {
+    val r = firstOf(_.
+      from(Account.Id(1)).on[Opened].event.
+      from(Transaction.Id(1)).on[Created].event)
+    "r : ProcessMonad[Created :+: Opened :+: CNil]" should compile
+  }
+
+  "ProcessSyntax.firstOf " should " allow to mix selectors and events from aggregates" in {
+    val r = firstOf(_.
+      from(Account.Id(1)).on[Opened].terminate.
+      on(selectorClosed).event.
+      from(Transaction.Id(1)).on[Created].event.
+      on(selectorOpened).event)
+    "r : ProcessMonad[Opened :+: Created :+: Closed :+: Unit :+: CNil]" should compile
+  }
+
+  "ProcessSyntax.firstOfUnified " should " accept a single selector and return the event type" in {
+    val r = firstOfUnified(_.
+      on(selectorOpened).event)
+    "r : ProcessMonad[Opened]" should compile
+  }
+
+  "ProcessSyntax.firstOfUnified " should " accept a single selector with a map and return the result type of the map" in {
+    val r = firstOfUnified(_.
+      on(selectorOpened).map(_.owner))
+    "r : ProcessMonad[String]" should compile
+  }
+
+  "ProcessSyntax.firstOfUnified " should " accept a two selector and return the base type of the event type" in {
+    val r = firstOfUnified(_.
+      on(selectorOpened).event.
+      on(selectorClosed).event)
+    "r : ProcessMonad[Account.Event with Product with Serializable]" should compile
+  }
+
+  "ProcessSyntax.firstOfUnified " should " accept a two selector mapping to the same type and return this type" in {
+    val r = firstOfUnified(_.
+      on(selectorOpened).map(_ ⇒ "hi").
+      on(selectorClosed).map(_ ⇒ "there"))
+    "r : ProcessMonad[String]" should compile
+  }
+
+  "ProcessSyntax.firstOf " should " accept a timeout" in {
+    val r = firstOf(_.
+      timeout(Instant.now)(terminate))
+    "r : ProcessMonad[Unit :+: CNil]" should compile
+  }
+
+  "ProcessSyntax.firstOf " should " accept a timeout and a selector" in {
+    val r = firstOf(_.
+      on(selectorClosed).event.
+      timeout(Instant.now)(terminate))
+    "r : ProcessMonad[Unit :+: Closed :+: CNil]" should compile
+  }
+
+  "ProcessSyntax.firstOf " should " accept a timeout that returns a value and a selector" in {
+    val r = firstOf(_.
+      on(selectorClosed).event.
+      timeout(Instant.now)(Monad[ProcessMonad].pure("timeout")))
+    "r : ProcessMonad[String :+: Closed :+: CNil]" should compile
+  }
+
+  "ProcessSyntax.firstOf(_.selector.event) " should " should use one selectors and return the event" in {
+    val evt = Opened("Mario")
+    firstOf(_.
+      on(selectorOpened).event) should runFromWithResult(
+      Expect.firstOf(FirstOfSelector(selectorOpened))(0, evt)
+    )(Coproduct[Opened :+: CNil](evt))
+  }
+
+  //TODO more tests
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Take the required action to fix the issue indicated by this comment.](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2603037564/issues/source?bid=3298199&fileBranchId=3298202#l222)